xc/xenops_server_xen.ml

```diff
@@ -2082,7 +2085,11 @@ module VBD = struct
 						)
 						(fun () ->
 							Opt.iter (fun domid ->
-								Storage.dp_destroy task (Storage.id_of (string_of_int domid) vbd.Vbd.id)
+								Opt.iter
```

Comment:
Just an observation: Would it be cleaner to use a flatter structure to identify the cases where something needs to happen? I feel `Opt.iter` is not helping here. Something like:

    match domid, device with
    | None, None ->
    | Some id, Some dev ->
    | None, Some dev ->
    | ...